### ISSUE TITLE
Finish search system

### ISSUE THREAD

**Issue by voltan** [pi-engine/pi#1437]
* Check new search engine chnages
* Update modules to use new engine

**Comment by voltan** [comment]
Finished, but more test needed

**Comment by Marc-pi** [comment]
* news / event : duplicated search result !!! 

![image](https://cloud.githubusercontent.com/assets/919332/22570780/0d96deae-e99d-11e6-84e3-6724dca30134.png)

* still old bugs (please have a look at our issue in guide issues)
   * not all results are retrieved (search holes), 
   * and result sort has to be fixed
   * location are not retrieved

**Comment by voltan** [comment]
> news / event : duplicated search result !!!

You need update search / event / news / guide modules to last version of git codes , all of them is update ?

> search result page search/guide/?q=keyword is uggly ! 

Add this css on theme please 
```
.search-result-item .thumbnail .caption {
    min-height: 80px;
}

> search result for page : wrong characters display in results, for latin characters éèàù, etc. !!!!

I dont have any idea for it, I think its your old issues of markup service , we use code like this : 
```
Pi::service("markup")->render($item["text_summary"], "html", "text");
```

> not all results are retrieved (search holes),

I see problem on search tab, did you have add any change on this part? when click on taps , search result not show, need back to old codes , that work true

> and result sort has to be fixed

we need add and update it on search api on all modules , I will do it 
```
    protected $order = array(
        'id DESC'
    );
```

> location are now well retrieved, but we need to isolate search results to distinguish category from location from item results

I can not undeastand wht do you want , if more customized needed I think we have to add custom search query on guide module , I still have problem by angolar js and have to add filter form manually on template

**Comment by voltan** [comment]
> we need add and update it on search api on all modules , I will do it

finished , please check order

**Comment by Marc-pi** [comment]
yes, everything was updated twice, core r5025 and modules 


-----
we you update ALL modules from system/modules/updateall

[warn] #20 in module/page/src/Api/Search.php
Declaration of Module\Page\Api\Search::buildUrl(array $item) should be compatible with Pi\Search\AbstractSearch::buildUrl(array $item, $table = '')

[warn] in module/comment/src/Api/Search.php
Declaration of Module\Comment\Api\Search::fetchResult(Pi\Application\Model\Model $model, Pi\Db\Sql\Where $where, $limit = 0, $offset = 0) should be compatible with Pi\Search\AbstractSearch::fetchResult(Pi\Application\Model\Model $model, Pi\Db\Sql\Where $where, $limit = 0, $offset = 0, $table = '')

[warn] in module/event/src/Api/Search.php
Declaration of Module\Event\Api\Search::getModel($table) should be compatible with Pi\Search\AbstractSearch::getModel($table = '')

[warn] in module/article/src/Api/Search.php
Declaration of Module\Article\Api\Search::buildUrl(array $item) should be compatible with Pi\Search\AbstractSearch::buildUrl(array $item, $table = '')

I'll answer your question below in another post

**Comment by Marc-pi** [comment]
> news / event : duplicated search result !!!

still there !!!!

> Add this css on theme please

can we add this in default guide css ?

> search result for page : wrong characters display in results, for latin characters éèàù, etc. !!!!

still there. why is it buggy only with Page module ???

> not all results are retrieved (search holes),

no, old code was buggy too !!!

> search result order
still old bugs (please have a look at our issue in guide issues)
 * not all results are retrieved (search holes),
 * and result sort has to be fixed
 * location are now well retrieved, but we need to isolate search results to distinguish category from location from item results

see https://github.com/pi-engine/guide/issues/387#issuecomment-273120064
and the comment below (fnac.com example i was talking over there about)

**Comment by Marc-pi** [comment]
in /search/?q=nous page

[warn] in module/event/src/Api/Search.php
Declaration of Module\Event\Api\Search::getModel($table) should be compatible with Pi\Search\AbstractSearch::getModel($table = '')
12:50:09.5451
[warn] #20 in module/page/src/Api/Search.php
Declaration of Module\Page\Api\Search::buildUrl(array $item) should be compatible with Pi\Search\AbstractSearch::buildUrl(array $item, $table = '')

**Comment by Marc-pi** [comment]
in search/event/?q=nous

[warn] #18 in module/event/src/Api/Search.php
Declaration of Module\Event\Api\Search::getModel($table) should be compatible with Pi\Search\AbstractSearch::getModel($table = '')

in search/page/?q=nous

[warn] #20 in module/page/src/Api/Search.php
Declaration of Module\Page\Api\Search::buildUrl(array $item) should be compatible with Pi\Search\AbstractSearch::buildUrl(array $item, $table = '')

**Comment by voltan** [comment]
please update core to my last push

**Comment by voltan** [comment]
> can we add this in default guide css ?

Yes but have to load guide css on search-resutl.phtml template too

**Comment by voltan** [comment]
please check by last version of codes , after that is any bug is still available we can fix it

**Comment by Marc-pi** [comment]
i still have those issues listed above :
* warning when updateALL in system module
* search holes, etc.

please recheck from https://github.com/pi-engine/pi/issues/1437#issuecomment-277438661

**Comment by voltan** [comment]
All of them?

**Comment by voltan** [comment]
I Update all modules and core to last version and check , I didnt see this errors

**Comment by Marc-pi** [comment]
Here I run PHP7.1

**Comment by voltan** [comment]
its not for php version , I had this errors on php 5.6 and fix all of them , please check modules and core is update , did you update `Lib/Pi/Search` from my last change? and last change of page module?

**Comment by Marc-pi** [comment]
i made a 3rd attempt, updating core and module, and i still have the warning when CLICKING ON UPDATE ALL MODULE, so on this page : /admin/system/module/refresh/active/1/from/installed

[warn] in module/comment/src/Api/Search.php
Declaration of Module\Comment\Api\Search::fetchResult(Pi\Application\Model\Model $model, Pi\Db\Sql\Where $where, $limit = 0, $offset = 0) should be compatible with Pi\Search\AbstractSearch::fetchResult(Pi\Application\Model\Model $model, Pi\Db\Sql\Where $where, $limit = 0, $offset = 0, $table = '')

[warn] in module/event/src/Api/Search.php
Declaration of Module\Event\Api\Search::getModel($table) should be compatible with Pi\Search\AbstractSearch::getModel($table = '')

[warn] in module/article/src/Api/Search.php
Declaration of Module\Article\Api\Search::buildUrl(array $item) should be compatible with Pi\Search\AbstractSearch::buildUrl(array $item, $table = '')

**Comment by Marc-pi** [comment]
and please fix search holes and result sort https://github.com/pi-engine/guide/issues/387#issuecomment-273120064

**Comment by voltan** [comment]
This 3 warnings fixed

**Comment by voltan** [comment]
And I made change on search result query , can you please check it and see result is better or not : https://github.com/pi-engine/pi/commit/0e5fff5a6a88a456b2fb253caef6ab18b230cc02

**Comment by voltan** [comment]
I think we can close it

**Comment by Marc-pi** [comment]
well, we will fix search result / search holes in another isssue
here,  we will have to improve layout consistency :
* display in all case a header well background  : display nr of result, or no result message
    * actually this block is displayed only in gloval search
* on guide, we must have consisten rounder images : design is different from global search vs guide search. 
* On global we have list result per raw, on guide search it is column display

------------

* Global Search

![image](https://cloud.githubusercontent.com/assets/919332/22653587/2c70a6e4-ec8a-11e6-93e9-e789a66c96e6.png)

* guide search

![image](https://cloud.githubusercontent.com/assets/919332/22653610/3a3bf850-ec8a-11e6-9e0e-09708f5a0831.png)

* news

![image](https://cloud.githubusercontent.com/assets/919332/22653624/46056dba-ec8a-11e6-9bae-520294e0ae0f.png)

* event

![image](https://cloud.githubusercontent.com/assets/919332/22653642/55143cdc-ec8a-11e6-97b3-35869ffcc9fa.png)

* page
![image](https://cloud.githubusercontent.com/assets/919332/22653650/5d2b0cc0-ec8a-11e6-9b2e-6eac8ed199a1.png)

**Comment by voltan** [comment]
Can we talk about search module first here and after that when main search system finished talk about just guide here https://github.com/pi-engine/guide/issues/387 ?

> display in all case a header well background : display nr of result, or no result message 

Added

> on guide, we must have consistent rounded images : design is different from global search vs guide search.

Set it by css on theme , its best way and very easy 

> On global we have list result per raw, on guide search it is column display

I add some changes , please check it

**Comment by Marc-pi** [comment]
> display in all case a header well background

ok, good, thanks

> guide not rounded images : on guide, we must have consistent rounded images : design is different from global search vs guide search.

no, it's a non sense to rebuild theme css on each project. we must have a consistent way accross modules.

> warnings on front user side of search module (there are no warnings anymore on admin side)

[notice] #39 in module/search/src/Block/Block.php
Undefined index: delay

> On global we have list result per raw, on guide search it is column display

still ko

> search result for page : wrong characters display in results, for latin characters éèàù, etc. !!!!

still ko

**Comment by Marc-pi** [comment]
@voltan let's finish this also  !

**Comment by voltan** [comment]
> no, it's a non sense to rebuild theme css on each project. we must have a consistent way accross modules.

Yes , But it should control by css, have general class and use it on all template, and just theme style load on all templates / pages , than its better put it on theme style. on shop module I forward more result on shop page, if it work true we can use it for other modules like guide and video and event

> [notice] #39 in module/search/src/Block/Block.php
Undefined index: delay

You need update search module. I change version number , hope your problem solve

> search result for page : wrong characters display in results, for latin characters éèàù, etc. !!!!

Here use escape too , like other parts , how do you solve escape problem on other modules? I replace escape by escape helper 

> still ko, we must think about using a consistent way : all columns for all modules, or row results for all.
On row list is better if we want to activate push to front fees.
we need to check on mobile also

My idea is , we need plan to inactive search result pages and forward result to modules , for modules like shop / guide / video / event we can do it , but for news / blog / page and ... we need use search front, I still dont know,

perhaps show result like google is better way and for more result forward user to module page

> conf > anonymous search interval : it's secs or miliseconds ? we have to indicate what is the unit

is better to set it 0 , I think its sec, some of my customers have problem by it and I set it to 0

> I add ajax search for block, search index page should update by angolarjs method
